packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -188,6 +196,40 @@ export class HardhatNode extends EventEmitter {
     return [common, node];
   }
 
+  private static _validateHardforks(
+    localHardfork: string,
+    forkBlockNumber: number | undefined,
+    common: Common,
+    remoteChainId: number
+  ): void {
+    if (!common.gteHardfork("spuriousDragon")) {
+      throw new InternalError(
```

Comment:
Throwing `ProviderError`s because this are thrown from a request, as the `HardhatNode` instance is created on the first RPC request.